src/ucp/wireup/wireup_cm.c

```diff
@@ -336,12 +326,18 @@ static unsigned ucp_cm_client_uct_connect_progress(void *arg)
     ucp_ep_config_key_t key;
     void* ucp_addr;
     size_t ucp_addr_size;
-    ucs_queue_head_t tmp_pending_queue;
     ucp_rsc_index_t cm_idx;
+    ucs_queue_head_t tmp_pending_queue;
     ucs_status_t status;
 
     UCS_ASYNC_BLOCK(&worker->async);
 
+    ucs_queue_head_init(&tmp_pending_queue);
+
+    /* Cleanup the previously created UCP EP. The one that was created on the
+     * previous call to this client's resolve_cb */
+    ucp_wireup_cm_ep_cleanup(ep, &tmp_pending_queue);
```

Comment:
why need to move?

Reply:
this needs to be done before configuration was changed to the new one